File: src/0083.remove-duplicates-from-sorted-list.83/solution.go
package removeduplicatesfromsortedlist83

type ListNode struct {
	Val  int
	Next *ListNode
}

func deleteDuplicates(head *ListNode) *ListNode {
	if head == nil {
		return head
	}
	head.Next = remove(head.Next, head.Val)
	return head
}

func remove(head *ListNode, prev int) *ListNode {
	if head == nil {
		return head
	}
	if head.Val == prev {
		return remove(head.Next, prev)
	}
	head.Next = remove(head.Next, head.Val)
	return head
}

File: src/0083.remove-duplicates-from-sorted-list.83/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 * int val;
 * ListNode next;
 * ListNode() {}
 * ListNode(int val) { this.val = val; }
 * ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        if (head == null) {
            return null;
        }
        ListNode last = head;
        ListNode current = head.next;
        while (current != null) {
            if (current.val == last.val) {
                current = current.next;
                last.next = current;
            } else {
                last = current;
                current = current.next;
            }

        }

        return head;
    }
}